source/iface/param_traits.rst

```diff
@@ -2,23 +2,197 @@
+interoperability within an application between SYCL and the
+associated SYCL backend API.
+
+There are three forms of interoperability with SYCL runtime classes:
+interoperability on the SYCL application with the SYCL backend API,
+interoperability within a SYCL kernel function with the equivalent
+kernel language types of the SYCL backend, and interoperability
+within a host task with the ``interop_handle``.
+
+SYCL application interoperability, SYCL kernel function interoperability
+and host task interoperability are provided via different interfaces
+and may have different behavior for the same SYCL object.
+
+SYCL application interoperability may be provided for ``buffer``, ``context``,
+``device``, ``device_image``, ``event``,`` kernel``, ``kernel_bundle``,
+``platform``, ``queue``, ``sampled_image``, and ``unsampled_image``.
+
+SYCL kernel function interoperability may be provided for ``accessor``,
+``device_event``, ``local_accessor``, ``sampled_image_accessor``, ``tream``
+and ``unsampled_image_accessor`` inside kernel scope only and
+is not available outside of that scope.
+
+host task interoperability may be provided for ``accessor``,
+``sampled_image_accessor``, ``unsampled_image_accessor``, ``queue``,
+``device``, ``context`` inside the scope of a host task only.
+
+.. seealso:: |SYCL_SPEC_HOST_TASK|
+
+
+
+.. backend_traits`:
 
+``sycl::backend_traits``
+========================
 ::
 
-  template <typename T, T param>
-  class param_traits;
 
-.. rubric:: Namespace
+  namespace sycl {
 
+  template <backend Backend> class backend_traits {
+   public:
+    template <class T> using input_type = /* see below */;
+
+    template <class T> using return_type = /* see below */;
+  };
+
+  template <backend Backend, typename SyclType>
+  using backend_input_t =
+      typename backend_traits<Backend>::template input_type<SyclType>;
+
+  template <backend Backend, typename SyclType>
+  using backend_return_t =
+      typename backend_traits<Backend>::template return_type<SyclType>;
+
+  } // namespace sycl
+
+A series of type traits are provided for SYCL backend interoperability,
+defined in the ``backend_traits`` class.
+
+A specialization of ``backend_traits`` must be provided for each named
+SYCL backend enumerated in the enum class ``backend`` that is
+available at compile time.
+
+The type alias ``backend_input_t`` is provided to enable less verbose access
+to the ``input_type`` type within ``backend_traits`` for a specific SYCL
+object of type ``T``. The type alias ``backend_return_t`` is provided to
+enable less verbose access to the ``return_type`` type within
+``backend_traits`` for a specific SYCL object of type ``T``.
+
+.. seealso:: |SYCL_SPEC_BCK_TRAITS|
+
+.. get_native`:
+
+``sycl::get_native``
+====================
 ::
 
-   info
 
-.. rubric:: Member types
+  namespace sycl {
+
+  template <backend Backend, class T>
+  backend_return_t<Backend, T> get_native(const T& syclObject);
+
+  } // namespace sycl
+
```

Comment:
I think it's worth including here the spec description of what `get_native` does - specifically:

> For each SYCL runtime class T which supports SYCL application interoperability, a specialization of get_native must be defined, which takes an instance of T and returns a SYCL application interoperability native backend object associated with syclObject which can be used for SYCL application interoperability. The lifetime of the object returned are backend-defined and specified in the backend specification.

> For each SYCL runtime class T which supports kernel function interoperability, a specialization of get_native must be defined, which takes an instance of T and returns the kernel function interoperability native backend object associated with syclObject which can be used for kernel function interoperability. The availability and behavior of these template functions is defined by the SYCL backend specification document.

Reply:
The text removed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/93/commits/2b756bc650da0d93cc364d2de99e2734c14d33ee